api/pkg/import/handler.go

```diff
@@ -0,0 +1,130 @@
+package _import
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/go-openapi/runtime/middleware"
+
+	"github.com/keptn/keptn/api/models"
+	"github.com/keptn/keptn/api/restapi/operations/import_operations"
+
+	logger "github.com/sirupsen/logrus"
+)
+
+const defaultImportArchiveExtension = ".zip"
+
+//go:generate moq -pkg fake --skip-ensure -out ./fake/projectchecker_mock.go . projectChecker:ProjectCheckerMock
+
+type projectChecker interface {
+	ProjectExists(projectName string) (bool, error)
+}
+
+// ImportHandler is the rest handler for the /import endpoint
+type ImportHandler struct {
+	checker                    projectChecker
+	tempStorageDir             string
+	maxUncompressedPackageSize uint64
+}
+
+// GetImportHandlerFunc will instantiate a configured ImportHandler and return the method that can be used for
+// handling http requests to the endpoint. See restapi.configureAPI for usage
+func GetImportHandlerFunc(storagePath string, checker projectChecker, maxPackageSize uint64) func(
+	params import_operations.ImportParams, principal *models.Principal,
+) middleware.Responder {
+	ih := getImportHandlerInstance(storagePath, checker, maxPackageSize)
+	return ih.HandleImport
+}
+
+func getImportHandlerInstance(storagePath string, checker projectChecker, maxPackageSize uint64) *ImportHandler {
+	return &ImportHandler{
+		checker:                    checker,
+		tempStorageDir:             storagePath,
+		maxUncompressedPackageSize: maxPackageSize,
+	}
+}
+
+// HandleImport is the method invoken when a POST request is received on the import endpoint.
+// This method will check that the project passed as parameter alredy exists in Keptn (
+// return a 404 immediately it that is not the case),
+// save the import package on the scratch storage and parse its contents.
+func (ih *ImportHandler) HandleImport(
+	params import_operations.ImportParams, principal *models.Principal,
+) middleware.Responder {
+
+	// Check if the project exists
+	projectExists, err := ih.checker.ProjectExists(params.Project)
+
+	if err != nil || !projectExists {
+		message := fmt.Sprintf("project %s does not exist", params.Project)
+		if err != nil {
+			message = fmt.Sprintf("error checking for project %s existence : %v", params.Project, err)
+		}
+
+		error := models.Error{
+			Code:    http.StatusNotFound,
```

Comment:
@christian-kreuzberger-dtx any thoughts ?

Reply:
@thisthat I've told @pchila to avoid that for internal API calls, as in the end it would mean having *multiple* HTTP Status 500 API Calls instead of just one.
For me, the end-result is what matters: We could not find the project, for whatever reason.

Reply:
I agree with your reasoning @christian-kreuzberger-dtx and we should avoid 500 errors here! 
I think we should instead return a 424 error as we have done for the automatic provisioning: https://github.com/keptn/keptn/issues/7149